CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/plugins/ExtractCoordsFromGraphPlugin.java

```diff
@@ -0,0 +1,235 @@
+
+    private static final double POINT_MARKER_X_OFFSET = 95;
+    private static final double POINT_MARKER_Y_OFFSET = 245;
+
+    private MapViewTopComponent mapViewTopComponent;
+
+    public ExtractCoordsFromGraphPlugin() {
+
+    }
+
+    public ExtractCoordsFromGraphPlugin(final MapViewTopComponent topComponent) {
+        mapViewTopComponent = topComponent;
+    }
+
+    @Override
+    public String getName() {
+        return "ExtractCoordsFromGraphPlugin";
+    }
+
+    /**
+     * Read the graph to extract geo coordinated
+     *
+     * @param graph - the current graph
+     * @param interaction
+     * @param parameters
+     * @throws InterruptedException
+     * @throws PluginException
+     */
+    @Override
+    protected void read(final GraphReadMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+
+        if (graph != null) {
+
+            final GraphElementType[] elementTypes = new GraphElementType[]{GraphElementType.VERTEX, GraphElementType.TRANSACTION};
+
+            mapViewTopComponent.getMapViewPane().getMap().clearQueriedMarkers();
+
+            try {
+                for (final GraphElementType elementType : elementTypes) {
+                    // Ids for all attributes needed from a single vertext of a graph
+                    int lonID = GraphConstants.NOT_FOUND;
+                    int latID = GraphConstants.NOT_FOUND;
+                    int colourID = GraphConstants.NOT_FOUND;
+                    int blazeID = GraphConstants.NOT_FOUND;
+                    int overlayID = GraphConstants.NOT_FOUND;
+                    int labelAttrID = GraphConstants.NOT_FOUND;
+                    int identifierID = GraphConstants.NOT_FOUND;
+
+                    int elementCount;
+
+                    switch (elementType) {
+                        case VERTEX:
+                            // Get IDs
+                            lonID = SpatialConcept.VertexAttribute.LONGITUDE.get(graph);
+                            latID = SpatialConcept.VertexAttribute.LATITUDE.get(graph);
+                            colourID = VisualConcept.VertexAttribute.COLOR.get(graph);
+                            blazeID = VisualConcept.VertexAttribute.BLAZE.get(graph);
+                            overlayID = VisualConcept.VertexAttribute.OVERLAY_COLOR.get(graph);
+                            labelAttrID = VisualConcept.VertexAttribute.LABEL.get(graph);
+                            identifierID = VisualConcept.VertexAttribute.IDENTIFIER.get(graph);
+                            elementCount = graph.getVertexCount();
+                            break;
+                        case TRANSACTION:
+                            lonID = SpatialConcept.VertexAttribute.LONGITUDE.get(graph);
+                            latID = SpatialConcept.VertexAttribute.LATITUDE.get(graph);
```

Comment:
I presume you want to use the transaction attribute here rather than the vertex one